controller/controller.go

```diff
@@ -270,9 +270,21 @@ func (c *HAProxyController) updateHAProxy() {
 			return c.Client.FrontendBindEdit("healthz",
 				models.Bind{
 					Name:    "v4",
-					Address: "*:1042",
+					Address: "0.0.0.0:1042",
 				})
 		}))
+
+		if !c.osArgs.DisableIPV6 {
+			logger.Panic(c.clientAPIClosure(func() error {
+				return c.Client.FrontendBindCreate("healthz",
+					models.Bind{
+						Name:    "v6",
+						Address: ":::1042",
+						V4v6:    true,
+					})
+			}))
+		}
+
```

Comment:
this is currently done in updateHAProxy which means  that this check will be run on every k8s update.

proper place for this is https://github.com/haproxytech/kubernetes-ingress/blob/a83d533b744d01e2fa1f470517343253b82d7529/controller/controller.go#L401

Reply:
actually sorry, this is ok :)